uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -477,6 +477,20 @@ private static function validateEntitlementAndWorkingDayAndBalanceChange($params
 
     $currentBalance = $leavePeriodEntitlement->getBalance();
 
+    if (!empty($params['id'])) {
+      $oldLeaveRequest = self::findById($params['id']);
+
+      if ($oldLeaveRequest && $oldLeaveRequest->id && self::isAlreadyApproved($oldLeaveRequest)) {
+        $periodContainingOldLeaveDates = self::getPeriodContainingDates($oldLeaveRequest);
+        $isSamePeriodWithOldLeave = $periodContainingOldLeaveDates->id == $period->id;
+
+        if ($isSamePeriodWithOldLeave) {
+          $oldLeaveRequestBalance = abs(LeaveBalanceChange::getTotalBalanceChangeForLeaveRequest($oldLeaveRequest));
+          $currentBalance += $oldLeaveRequestBalance;
+        }
+      }
+    }
+
```

Comment:
This is basically the same problem from PCHR-2471. I feel that a better approach would be to change LeavePeriodEntitlement.getBalance() to also, optionally, accept the ID of a Leave Request to be excluded from the balance. 

The `$leavePeriodEntitlement` instance is only related to the period of containing the from_date and to_date in $params, right? If that's true, then:
- If the dates didn't change, we'll get the balance containing all the approved leave requests, except the one with the given ID, and then the `$leaveRequestBalance > $currentBalance` will work
- If the dates changed, but are still in the same period, the same logic above will work
- If the dates changed and are now in a different period, `$leavePeriodEntitlement` will refer to this new period and will try to exclude the balance changes for the given leave request, but, on the DB, the leave request still belongs to the old period and will not be included, i.e. the logic will still work

There's another problem here though. Currently, we're always using the `calculateBalanceChangeFromCreateParams()` method to get the balance change for the Leave Request. If there are changes to the work pattern(s) used during the request period, it can cause the balance to be different than what it was when the request was approved. This should be consistent with the new `change_balance` logic. If that param is present and true, then we calculate the balance change according to the new data, otherwise, we consider the value we have in the DB